### PR TITLE
initialise testing setup

### PR DESCRIPTION
**Pull request by m4v15**
Just need to think about adding test db etc

relates #47

### CODE REVIEW

**Comment by mattlub** [comment]
good stuff so far m4te

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+language: node_js
+node_js:
+  - "v7"
```
Comment (mattlub) [inline .travis.yml]: v6?

**Review by mattlub — CHANGES_REQUESTED** [review]

**Comment by m4v15** [comment]
[Tape issue](https://github.com/substack/tape/issues/216#issuecomment-314717299)

**Review by mattlub — APPROVED** [review]

**Comment by des-des** [comment]
```js
require('../config.js')
const test = require('tape')
const dbConnection = require('../db/connect.js')
// const server = require('../src/server.js')
// const supertest = require('supertest')
// const Users = require('../db/models/User.js')
// ^ will use these in the futue
test('Check tape is working before db connection', t => {
  t.equal(1, 1, 'One is one')
  t.end()
})

test('Check db connection', t => {
  t.plan(1)
  dbConnection.once('open', () => {
    t.ok(true)
  })
})

dbConnection.on('error', err => {
  throw err
})

test.onFinish(() => {
  dbConnection.close()
})
```

File: config.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  if (process.env.NODE_ENV === 'test') {
+    // set environment variables for test environment
+    process.env.MONGODB_URI = process.env.MONGODB_URI_TEST
+  } else if (process.env.NODE_ENV === 'development') {
+    // set environment variables for dev environment
+    process.env.MONGODB_URI = process.env.MONGODB_URI_DEV
+  }
+}
+
+console.log(process.env.MONGODB_URI)
```
Comment (des-des) [inline config.js]: :fire:

File: config.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+if (process.env.NODE_ENV !== 'production' && !process.env.CI) {
+  require('dotenv').config()
+  if (process.env.NODE_ENV === 'test') {
+    // set environment variables for test environment
```
Comment (des-des) [inline config.js]: I think this is self explanatory

**Review by des-des — CHANGES_REQUESTED** [review]
Update readme

mongo,
.env,
scripts in package

File: config.js (hunk 2 above)
Comment (m4v15) [reply config.js]: oops sorry...

**Comment by des-des** [comment]
@m4v15 @mattlub dropped this into the gitter but gonna but it here also
```js
// index.test.js
const test = require('tape')

dbConnection.once('open', () => {
  require('./test_1.test.js')
  require('./test_1.test.js')

  test.onFinish(() => {
    dbConnection.close()
  })
}
// Do not forget this!
dbConnection.on('error', err => {
  throw err
})

// test_1.test.js
const test = require('tape')

tape('this is the first test', t => {
  // do what you want, async, t.plan, t.end, whatever
})
```

**Comment by des-des** [comment]
@m4v15 @mattlub dropped this into the gitter but gonna but it here also
```js
// index.test.js
const test = require('tape')

dbConnection.once('open', () => {
  require('./test_1.test.js')
  require('./test_1.test.js')

  test.onFinish(() => {
    dbConnection.close()
  })
}
// Do not forget this!
dbConnection.on('error', err => {
  throw err
})

// test_1.test.js
const test = require('tape')

tape('this is the first test', t => {
  // do what you want, async, t.plan, t.end, whatever
})
```

**Review by des-des — APPROVED** [review]

**Comment by des-des** [comment]
@m4v15 sweet, assign me for merge!

**Comment by m4v15** [comment]
@des-des just adding the readme stuff

**Pull request merged**